evalml/automl/automl_search.py

```diff
@@ -251,6 +251,7 @@ def __init__(self,
             'pipeline_results': {},
             'search_order': [],
         }
+        self._pipelines_searched = dict()
```

Comment:
Using this to store pipeline instances rather than in `results` because we don't want to deepcopy (lightgbm prints lots of output when deepcopied)